microprofile/grpc/core/src/main/java/io/helidon/microprofile/grpc/core/Instance.java

```diff
@@ -129,15 +132,28 @@ public T get() {
             return ensureInstance();
         }
 
-        private synchronized T ensureInstance() {
-            if (instance == null) {
-                try {
-                    instance = instanceClass.newInstance();
-                } catch (Throwable e) {
-                    throw Status.INTERNAL.withCause(e).asRuntimeException();
+        private T ensureInstance() {
+            return accessInstance(() -> {
+                if (instance == null) {
+                    try {
+                        instance = instanceClass.newInstance();
+                    } catch (Throwable e) {
+                        throw Status.INTERNAL.withCause(e).asRuntimeException();
+                    }
                 }
+                return instance;
+            });
+        }
+
+        private <T> T accessInstance(Supplier<T> operation) {
+            try {
+                instanceAccess.acquire();
+                T result = operation.get();
+                instanceAccess.release();
```

Comment:
This will not happen if `operation.get()` fails.

Reply:
Fixed